### PR TITLE
Add search page

### PR DESCRIPTION
**Pull request by rushenn**
- Default Sphinx search page has been replaced with Algolia's 
- Enabled routing to sync the search results with URL

**Preview:**
Screenshot 2022-12-30 at 14 47 28

**How to test:**
1. Click on a search result, navigate back using the browser back button and check whether the search result modal is open and available.
2. Type a search keyword and press the enter key to navigate to the search results page, where you can find all the results listed. 

Note: You might need to start the local web server on the exact build sub-directory to test the point number 2.  
`build/[branch]/[platform]/html/` - ✔️
`build/` - ❌

### CODE REVIEW

**Comment by djwfyi** [comment]
GREAT WORK, @rushenn! This is awesome.
I'm merging it now, but would like to see an H1 or H2 title on the search page.

File: source/_templates/search.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            <a class="sidebar__title" href="{{ pathto('index') }}">{{ shorttitle}}</a>
+            
+            {{ miniosidebar() }}
+         </div>
+
+         {%- include "toc.html" %}
+
+         <div class="content__main" id="search-documentation">
+            {% block body %} 
+               <div id="search">
```
Comment (djwfyi) [inline source/_templates/search.html:53]: Can we add a page title here?

# Search

or

# Docs Search

**Review by djwfyi — APPROVED** [review]
LGTM
Merging now, but do have one request for the search page.

File: source/_templates/search.html (hunk 1 above)
Comment (rushenn) [reply source/_templates/search.html:53]: Sure @djwfy, I will send another to add the title.

**Pull request merged**